src/components/SortByForm.jsx

```diff
@@ -1,42 +1,92 @@
-import { useEffect, useState } from "react"
-import { getCategories } from "../axios"
-import { useNavigate } from "react-router-dom"
+import { useEffect, useState } from "react";
+import { getCategories } from "../axios";
+import { useNavigate } from "react-router-dom";
 
-export default function SortByForm ({ setOrderBy, setCategory, category }) {
-    const [ filterCategories, setFilterCategories ] = useState([])
-    const navigate = useNavigate()
+export default function SortByForm({
+  setOrderBy,
+  orderBy,
+  setCategory,
+  category,
+  setOrderDirection,
+  orderDirection,
+}) {
+  const [filterCategories, setFilterCategories] = useState([]);
+  const [ selectedOrder , setSelectedOrder ] = useState("")
+  const [ selectedDirection, setSelectedDirection ] = useState("")
 
-    useEffect(() => {
-        getCategories()
-        .then(({ data }) => {
-            setFilterCategories(data.categories)
-        })
-    }, [])
+  const navigate = useNavigate();
 
-    const handleFilterClick = (e) => {
-        e.preventDefault()
-        navigate(category)
-    }
+  useEffect(() => {
+    getCategories().then(({ data }) => {
+      setFilterCategories(data.categories);
+    });
+  }, []);
 
-    return (
-        <form className="sort-by-form">
-            <label htmlFor="category-filter">Filter by Category</label>
-            <select onClick={e => setCategory(e.target.value)} name="category" id="category-filter">
-                <option value="" selected>all</option> 
-                {filterCategories.map((category, i) => {
-                    const categoryName = category.slug.replaceAll("-", " ")
-                    return <option key={i} value={categoryName}>{categoryName}</option>
-                })}   
-            </select> 
-            <button onClick={handleFilterClick}>Filter</button><br />
-            <label htmlFor="sort-options">Order By</label>
-            <select name="sort" id="sort-options">
-                <option value="created_at">date</option>
-                <option value="title">title</option>
-                <option value="owner">review author</option>
-                <option value="votes">number of votes</option>
-            </select>
-            <button>Descending</button>
-        </form>
-    )
-}
\ No newline at end of file
+  const handleFilterClick = (e) => {
+    e.preventDefault();
+    navigate(category);
+  };
+
+  const handleOrderClick = (e) => {
+    e.preventDefault();
+    setOrderBy(selectedOrder)
+    setOrderDirection(selectedDirection)
+  };
+
+  return (
+    <form className="sort-by-form">
+      <label htmlFor="category-filter">Filter by Category</label>
+      <select
+        onClick={(e) => setCategory(e.target.value)}
```

Comment:
```js
        onClick={(e) => setCategory(e.target.value)}
```
This should be `onChange` so that it sets the state after they've picked a value. onClick will fire when they open the list and will set the state to the current option, not the new one.